File: 03-LongestWord/3.hs
--Find the longest word in a sentence while excluding punctuation
import Data.List

longestWord :: String -> String
longestWord str = head . sortBy wordLen . map rmvPunct . words $ str
  where wordLen x y = compare (length y) (length x) 
        rmvPunct = filter (`notElem` ".?!-;\"@#$%^&*():,/\'{}[]")

main = do
  print $ longestWord "i love$..! bacon" == "bacon"

File: 03-LongestWord/3.js
function LongestWord(sen) { 
  function rmPunct(s){ return s.replace(/[^\w\s]/g,"")}

  var words = sen.split(' ').sort(function(a,b){
    return rmPunct(b).length-rmPunct(a).length
  })
  return rmPunct(words[0])
}

console.log(LongestWord("I love!!!...$$$ frosting.") == "frosting")